clients/src/main/java/org/apache/kafka/common/requests/TxnOffsetCommitRequest.java

```diff
@@ -93,7 +118,8 @@ public TxnOffsetCommitRequest build(short version) {
                 throw new UnsupportedVersionException("Broker doesn't support group metadata commit API on version " + version
                 + ", minimum supported request version is 3 which requires brokers to be on version 2.5 or above.");
             }
-            return new TxnOffsetCommitRequest(data, version);
+
+            return new TxnOffsetCommitRequest(data, (short) Math.min(desiredMaximumVersion, version));
```

Comment:
Why do we need this for this request but not the produce request?

Reply:
Added the same thing to the produce request.